src/dune_cache/config.ml

```diff
@@ -3,25 +3,42 @@ open Stdune
 (* CR-someday amokhov: We should probably switch from float [check_probability]
    to integer [check_frequency], as in Jenga, to avoid generating random floats. *)
 
-(* CR-soon amokhov: That that the reproducibility check actually works. *)
-
 module Reproducibility_check = struct
   (* CR-someday amokhov: Add [Check_and_repair] to rewrite cache entries if they
      disagree with the check. *)
   type t =
     | Skip
-    | Check of { check_probability : float }
+    | Check_with_probability of float
+    | Check
```

Comment:
What's the difference between this constructor and `Check_with_probability 1.0`?

Reply:
I'm being a bit paranoid here and treat the probabilities 0 and 1 specially because in these two extremes the behaviour can be completely deterministic: we don't need to generate a random number and always either do or don't do the check.

In particular, this means that our testsuite is completely deterministic: we do not generate any random numbers when running it. If we used a plain `float` instead, we would generate random numbers and compare them with 0 or 1, which is mildly disturbing.

I'll add a comment to clarify this.

Reply:
Done.

Reply:
https://www.youtube.com/watch?v=ZA4JkHKZM50

Reply:
@snowleopard, in addition to being "mildly disturbing", the consumption of entropy from the random number generator can be an observable side-effect sometimes.

Reply:
Indeed!